.github/workflows/main-build.yml

```diff
@@ -102,12 +110,204 @@ jobs:
           name: aws-opentelemetry-agent.jar
           path: otelagent/build/libs/aws-opentelemetry-agent-*.jar
 
+      - name: Get ADOT_JAVA_AGENT Image Tag
+        id: java_agent_versioning
+        working-directory: aws-otel-java-instrumentation
+        run: |
+          shortsha="$(git rev-parse --short HEAD)"
+          java_agent_tag=${{ env.ADOT_JAVA_VERSION }}-$shortsha
+          echo "STAGING_TAG=java_agent_tag" >> $GITHUB_OUTPUT
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2.1.0
+        with:
+          role-to-assume: ${{ secrets.JAVA_INSTRUMENTATION_SNAPSHOT_ECR }}
+          aws-region: us-west-2
+
+      - name: Login to private staging ecr
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.STAGING_ECR }}
+        env:
+          AWS_REGION: us-west-2
+
+      - name: Build and push staging image for e2e integration test
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          build-args: "ADOT_JAVA_VERSION=${{ env.ADOT_JAVA_VERSION }}"
+          context: .
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            ${{ env.STAGING_ECR }}:${{ steps.java_agent_versioning.outputs.STAGING_TAG }}
+
+  create-test-ref:
+    runs-on: ubuntu-latest
+    outputs:
+      testRef: ${{ steps.setRef.outputs.ref }}
+    steps:
+      - name: Set testRef output
+        id: setRef
+        run: |
+          if [[ ${{ github.ref_name }} == release/v* ]]; then 
+            echo "ref=${{github.ref_name}}" >> $GITHUB_OUTPUT
```

Comment:
aren't we going to use a branch specifically for java. If you keep like this, there will be collisions with the collector.

my proposal was to use something like `java-release/v*`

Reply:
Thanks @rapphil. Updated.

Reply:
@rapphil thanks. Updated.